Source/WTF/wtf/generic/WorkQueueGeneric.cpp

```diff
@@ -41,16 +41,14 @@ WorkQueueBase::WorkQueueBase(RunLoop& runLoop)
 
 void WorkQueueBase::platformInitialize(const char* name, Type, QOS qos)
 {
-    BinarySemaphore semaphore;
-    Thread::create(name, [&] {
+    m_runLoop = RunLoop::create(name, ThreadType::Unknown, qos).ptr();
 #if ASSERT_ENABLED
+    BinarySemaphore semaphore;
+    m_runLoop->dispatch([&] {
         m_threadID = Thread::current().uid();
-#endif
-        m_runLoop = &RunLoop::current();
-        semaphore.signal();
```

Comment:
Is it intentional that signaling the semaphore is removed, while keeping the wait?

Reply:
Thanks!
(Appears that runloop tests aren't run WorkQueueGeneric platforms?)